Enable macOS previews in SwiftUI MiniBrowser BrowserTab

stwrt opened the pull request:
#### 3c59e3a33892b111ed0d9d8ffb21996a418e7c20
<pre>
Enable macOS previews in SwiftUI MiniBrowser BrowserTab
<a href="https://bugs.webkit.org/show_bug.cgi?id=258263">https://bugs.webkit.org/show_bug.cgi?id=258263</a>
Reviewed by Jonathan Bedard.

BrowserTab previews are available on iOS. This adds support to view them on macOS too.

* Tools/MiniBrowserSwiftUI/Shared/BrowserTab.swift:
(BrowserTab_Previews.previews):

Canonical link: <a href="https://commits.webkit.org/265493@main">https://commits.webkit.org/265493@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4555f050b3ae1aae17f3f7a37bb1325464e282fd

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/10506 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/10733 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/11008 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/12146 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/10076 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/10518 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/13097 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/10689 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/13017 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/10665 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/11599 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/8835 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/12545 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/8893 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/9491 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/16751 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/9964 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/9643 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/12893 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/10097 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/8184 "1 flakes 4 failures") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/9256 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/13513 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/1239 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/9959 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

JonWBedard commented on Tools/MiniBrowserSwiftUI/Shared/BrowserTab.swift:
```diff
@@ -160,7 +160,9 @@ private struct ExternalURLNavigation : Identifiable, Hashable {
 struct BrowserTab_Previews: PreviewProvider {
     static var previews: some View {
-#if os(iOS)
+#if os(macOS)
```
Does this not also disable previews on iOS?

stwrt replied:
iOS will be handled by the `#else` portion.

stwrt committed:
Enable macOS previews in SwiftUI MiniBrowser BrowserTab
https://bugs.webkit.org/show_bug.cgi?id=258263
Reviewed by Jonathan Bedard.

BrowserTab previews are available on iOS. This adds support to view them on macOS too.

* Tools/MiniBrowserSwiftUI/Shared/BrowserTab.swift:
(BrowserTab_Previews.previews):

Canonical link: https://commits.webkit.org/265493@main
